### PR TITLE
Tx gas limit

### PR DESCRIPTION
**Pull request by lsebrie**
Transactions shouldn't be able to send any value as gas limit. As a initial cap it will be limited to 2^63 -1 as the maximum value allowed.

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 159 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 106 major
* ![MINOR][MINOR] 38 minor
* ![INFO][INFO] 14 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Constants.java#L37](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/org/ethereum/config/Constants.java#L37): Use "BigDecimal.valueOf" instead. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2111)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L98](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L98): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L108](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L108): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L110](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L110): Method co.rsk.rpc.modules.eth.EthModuleWalletEnabled.accounts() passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L122](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L122): Method co.rsk.rpc.modules.eth.EthModuleWalletEnabled.sign(String, ECKey) concatenates the result of a toString() call [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AISB_TOSTRING_APPENDING)
1. ![MAJOR][MAJOR] [Constants.java#L70](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/org/ethereum/config/Constants.java#L70): Remove this method and declare a constant for this value. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS3400)
1. ![MAJOR][MAJOR] [Web3Impl.java#](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#): This file has 1,174 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Web3Impl.java#L81](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L81): Make repository a static final constant or non-public and provide accessors if needed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
1. ![MAJOR][MAJOR] [Web3Impl.java#L83](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L83): Make eth a static final constant or non-public and provide accessors if needed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
1. ![MAJOR][MAJOR] [Web3Impl.java#L110](https://github.com/rsksmart/rskj/blob/3cefcfc3ddc7a14e9778322cbfd50e0776d9e557/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L110): Constructor has 11 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by tinchou** [comment]
I just realized:

```java
@Test
public void numberTests() {
    BigInteger a = new BigDecimal(Math.pow(2, 63)).subtract(new BigDecimal(1)).toBigInteger();
    BigInteger b = BigInteger.valueOf(9_223_372_036_854_775_807L);
    BigInteger c = BigInteger.valueOf(Long.MAX_VALUE);
    assertEquals(a, b);
    assertEquals(a, c);
    assertEquals(b, c);
}
```

I think what we really want is this number instead, because it takes into account negative `long` values:

```java
@Test
public void otherNumberTests() {
    BigInteger a = new BigDecimal(Math.pow(2, 62)).subtract(new BigDecimal(1)).toBigInteger();
    BigInteger b = BigInteger.valueOf((1L << 62) - 1);
    assertEquals(a, b);
}
```

@SergioDemianLerner can you confirm??

File: rskj-core/src/test/java/org/ethereum/rpc/Web3ImplTest.java [hunk 1]
```diff
@@ -1230,6 +1256,44 @@ public void eth_sendTransaction()
...
+        String txHash = null;
+        try {
+            txHash = web3.eth_sendTransaction(args);
+        }
+        catch (JsonRpcInvalidParamException e) {
+            Assert.assertEquals(e.getMessage(), "Gas limit exceeds Transaction max value tolerated");
+        } catch (Exception e) {
+            Assert.assertFalse(e instanceof JsonRpcInvalidParamException);
+        }
+        Assert.assertNull(txHash);
```
Comment (ajlopez) [inline rskj-core/src/test/java/org/ethereum/rpc/Web3ImplTest.java]: could we use an Assert.fail? And then, in the first catch, add a return command, then the second catch could be removed

**Comment by rskops** [comment]
SonarQube analysis reported 132 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 106 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 14 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Constants.java#L38](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/org/ethereum/config/Constants.java#L38): Use "BigDecimal.valueOf" instead. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2111)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L101](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L101): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L111](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L111): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L113](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L113): Method co.rsk.rpc.modules.eth.EthModuleWalletEnabled.accounts() passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L125](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L125): Method co.rsk.rpc.modules.eth.EthModuleWalletEnabled.sign(String, ECKey) concatenates the result of a toString() call [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AISB_TOSTRING_APPENDING)
1. ![MAJOR][MAJOR] [Constants.java#L71](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/org/ethereum/config/Constants.java#L71): Remove this method and declare a constant for this value. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS3400)
1. ![MAJOR][MAJOR] [Web3Impl.java#](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#): This file has 1,195 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Web3Impl.java#L81](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L81): Make repository a static final constant or non-public and provide accessors if needed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
1. ![MAJOR][MAJOR] [Web3Impl.java#L83](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L83): Make eth a static final constant or non-public and provide accessors if needed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
1. ![MAJOR][MAJOR] [Web3Impl.java#L110](https://github.com/rsksmart/rskj/blob/1751e3b7a94edb92f82eb9042fef79856f5ce3ea/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L110): Constructor has 11 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/test/java/org/ethereum/rpc/Web3ImplTest.java (hunk 1 above)
Comment (lsebrie) [reply rskj-core/src/test/java/org/ethereum/rpc/Web3ImplTest.java]: I'll improve this so it's a bit cleaner

File: rskj-core/src/test/java/org/ethereum/rpc/Web3ImplTest.java [hunk 2]
```diff
@@ -295,6 +296,29 @@ public void sendRawTransaction() throws Exception {
...
+                .gasLimit(new BigDecimal(Math.pow(2, 60)).add(new BigDecimal(1)).toBigInteger()).build();
+
+        String rawData = Hex.toHexString(tx.getEncoded());
+
+        try {
+            web3.eth_sendRawTransaction(rawData);
+        } catch (JsonRpcInvalidParamException e) {
+                Assert.assertEquals(e.getMessage(), "Gas limit exceeds Transaction max value tolerated");
+        } catch (Exception e) {
+            Assert.fail();
```
Comment (ajlopez) [inline rskj-core/src/test/java/org/ethereum/rpc/Web3ImplTest.java]: In the above code, the test could pass WITHOUT raising exception. My suggestion: only one catch, AND a return at the end of the catch. THEN AFTER closing the try, put an Assert.fail(). In this way, it is clear that if the catch is not executed, then test FAILS

```
try {
    web3.eth_sendRawTransaction(rawData);
 } catch (JsonRpcInvalidParamException e) {
    Assert.assertEquals(e.getMessage(), "Gas limit exceeds Transaction max value tolerated");
	return;
 }

Assert.fail(); 
```

Comment (lsebrie) [reply rskj-core/src/test/java/org/ethereum/rpc/Web3ImplTest.java]: Yes it should check that always an exception is thrown.

**Comment by rskops** [comment]
SonarQube analysis reported 130 issues
* ![MAJOR][MAJOR] 105 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 14 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L101](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L101): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L111](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L111): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L113](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L113): Method co.rsk.rpc.modules.eth.EthModuleWalletEnabled.accounts() passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L125](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L125): Method co.rsk.rpc.modules.eth.EthModuleWalletEnabled.sign(String, ECKey) concatenates the result of a toString() call [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AISB_TOSTRING_APPENDING)
1. ![MAJOR][MAJOR] [Web3Impl.java#](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#): This file has 1,195 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Web3Impl.java#L81](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L81): Make repository a static final constant or non-public and provide accessors if needed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
1. ![MAJOR][MAJOR] [Web3Impl.java#L83](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L83): Make eth a static final constant or non-public and provide accessors if needed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
1. ![MAJOR][MAJOR] [Web3Impl.java#L110](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L110): Constructor has 11 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Web3Impl.java#L141](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L141): Constructor new org.ethereum.rpc.Web3Impl(Ethereum, WorldManager, RskSystemProperties, MinerClient, MinerServer, PersonalModule, EthModule, ChannelManager, Repository, PeerScoringManager, PeerServer) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)
1. ![MAJOR][MAJOR] [Web3Impl.java#L148](https://github.com/rsksmart/rskj/blob/8175b7a1da5b9dc4e129fb1036ec909c8661e72c/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L148): Reduce this anonymous class number of lines from 22 to at most 20, or make it a named class. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1188)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 130 issues
* ![MAJOR][MAJOR] 105 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 14 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L101](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L101): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L111](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L111): Extract the assignment out of this expression. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L113](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L113): Method co.rsk.rpc.modules.eth.EthModuleWalletEnabled.accounts() passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [EthModuleWalletEnabled.java#L125](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleWalletEnabled.java#L125): Method co.rsk.rpc.modules.eth.EthModuleWalletEnabled.sign(String, ECKey) concatenates the result of a toString() call [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AISB_TOSTRING_APPENDING)
1. ![MAJOR][MAJOR] [Web3Impl.java#](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#): This file has 1,195 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Web3Impl.java#L81](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L81): Make repository a static final constant or non-public and provide accessors if needed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
1. ![MAJOR][MAJOR] [Web3Impl.java#L83](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L83): Make eth a static final constant or non-public and provide accessors if needed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
1. ![MAJOR][MAJOR] [Web3Impl.java#L110](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L110): Constructor has 11 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Web3Impl.java#L141](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L141): Constructor new org.ethereum.rpc.Web3Impl(Ethereum, WorldManager, RskSystemProperties, MinerClient, MinerServer, PersonalModule, EthModule, ChannelManager, Repository, PeerScoringManager, PeerServer) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)
1. ![MAJOR][MAJOR] [Web3Impl.java#L148](https://github.com/rsksmart/rskj/blob/ae00b06db1f2f1a3fd0308da1dc2c9576978ae79/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L148): Reduce this anonymous class number of lines from 22 to at most 20, or make it a named class. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1188)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**